### PR TITLE
fix: LSDV-4684: Fix check for image boundaries

### PR DESCRIPTION
**Pull request by hlomzik**
When DEV-3793 FF is on this check is broken and user is able to create bbox going behind the right and bottom edges of the image.
It has to check against 100 instead of real sizes with FF.

### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
- [ ] Product design
- [x] Frontend

### Describe the reason for change
With FF on we should use relative sizes almost everywhere, including bbox creation.

#### What does this fix?
With FF on checks that bbox sizes are not exeed 100.

#### What is the new behavior?
_(if this is a breaking or feature change)_

#### What is the current behavior?
_(if this is a breaking or feature change)_

#### What libraries were added/updated?
_(list all with version changes)_

#### Does this change affect performance?
_(if so describe the impacts positive or negative)_

#### Does this change affect security?
_(if so describe the impacts positive or negative)_

#### What alternative approaches were there?
_(briefly list any if applicable)_

#### What feature flags were used to cover this change?
Fix is fixing issue with `fflag_fix_front_dev_3793_relative_coords_short`

### Does this PR introduce a breaking change?
_(check only one)_
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
_(check all that apply)_
- [ ] e2e
- [ ] integration
- [ ] unit

### Which logical domain(s) does this change affect?
_(for bug fixes/features, be as precise as possible. ex. Authentication, Annotation History, Review Stream etc.)_

### CODE REVIEW

File: e2e/tests/image.test.js [hunk 1]
```diff
@@ -138,3 +147,129 @@ Scenario('Image with perRegion tags', async function({ I, AtImageView, AtSidebar
...
+    data: { image },
+    task: {
+      id: 0,
+      annotations: [{ id: 1001, result: [] }],
+      predictions: [],
+    },
+  });
+
+  await AtImageView.lookForStage();
+  await AtImageView.waitForImage();
```
Comment (Gondragos) [inline e2e/tests/image.test.js:169]:
```suggestion
  await AtImageView.waitForImage();
  await AtImageView.lookForStage();
```

File: src/mixins/DrawingTool.js [hunk 2]
```diff
@@ -231,15 +231,28 @@ const TwoPointsDrawingTool = DrawingTool.named('TwoPointsDrawingTool')
...
+        let { x1, y1, x2, y2 } = isEllipse ? {
+          x1: shape.startX,
+          y1: shape.startY,
+          x2: x,
+          y2: y,
+        } : Utils.Image.reverseCoordinates({ x: shape.startX, y: shape.startY }, { x, y });
 
         x1 = Math.max(0, x1);
         y1 = Math.max(0, y1);
         x2 = Math.min(stageWidth, x2);
```
Comment (Gondragos) [inline src/mixins/DrawingTool.js:245]: We still need 
```js
x2 = Math.min(isFF(FF_DEV_3793) ? 100 : stageWidth, x2);
y2 = Math.min(isFF(FF_DEV_3793) ? 100 : stageHeight, y2);
```
for rectangles. The first test fails without it.

File: src/mixins/DrawingTool.js [hunk 3]
```diff
@@ -231,15 +231,28 @@ const TwoPointsDrawingTool = DrawingTool.named('TwoPointsDrawingTool')
...
         y2 = Math.min(stageHeight, y2);
 
-        shape.setPositionInternal(x1, y1, x2 - x1, y2 - y1, shape.rotation);
+        let [distX, distY] = [x2 - x1, y2 - y1].map(Math.abs);
+
+        if (isEllipse) {
+          distX = Math.min(distX, Math.min(x1, 100 - x1));
+          distY = Math.min(distY, Math.min(y1, 100 - y1));
```
Comment (Gondragos) [inline src/mixins/DrawingTool.js:252]:
```suggestion
          distX = Math.min(distX, Math.min(x1, (isFF(FF_DEV_3793) ? 100 : stageWidth) - x1));
          distY = Math.min(distY, Math.min(y1, (isFF(FF_DEV_3793) ? 100 : stageHeight) - y1));
```
This is a fix to keep it work without enabled FF_DEV_3793. I suppose it also fix some failing tests.

File: e2e/tests/image.test.js [hunk 4]
```diff
@@ -138,3 +147,129 @@ Scenario('Image with perRegion tags', async function({ I, AtImageView, AtSidebar
...
```
Comment (Gondragos) [inline e2e/tests/image.test.js:230]:
```suggestion
  await AtImageView.waitForImage();
  await AtImageView.lookForStage();
```

File: e2e/tests/image.test.js [hunk 5]
```diff
@@ -138,3 +147,129 @@ Scenario('Image with perRegion tags', async function({ I, AtImageView, AtSidebar
   assert.strictEqual(result.length, 1);
   assert.deepStrictEqual(result[0].value.rectanglelabels, ['Moonwalker']);
 });
+
+Scenario('Can\'t create rectangles outside of canvas', async ({ I, AtLabels, AtSidebar, AtImageView, LabelStudio }) => {
```
Comment (Gondragos) [inline e2e/tests/image.test.js:151]: We may want to use `DataTable` to save ourselve from the regression without relative coordinates.
Something like:
```js
const ffs = new DataTable(['FF_3793']);

ffs.add([true]);
ffs.add([false]);

Data(ffs).Scenario('Can\'t create rectangles outside of canvas', async ({ I, AtLabels, AtSidebar, AtImageView, LabelStudio, current }) => {
  const { FF_3793 } = current;

  LabelStudio.setFeatureFlags({
    fflag_fix_front_dev_3793_relative_coords_short: FF_3793,
  });
```
and reuse it for the second test. As I see it passes the tests without any other changes if there is everithing ok.

**Review by nicholasrq — APPROVED** [review]

**Pull request merged**